dashboard/config/routes.rb

```diff
@@ -1,6 +1,14 @@
 # For documentation see, e.g., http://guides.rubyonrails.org/routing.html.
 
 Dashboard::Application.routes.draw do
+  get 'teacher_dashboard/sections/:section_id/', to: 'teacher_dashboard#index'
```

Comment:
these could all condense to:

```
get 'teacher_dashboard/sections/:section_id/:tab', to: 'teacher_dashboard#index'
```

then, in the controller, we have access to `params[:tab]`, so the controller could have:

```
def index
  @tab = params[:tab]
end
```

then, in the view, we could supply the tab directly to our component via data (i don't remember the exact format for data objects, so don't quote me on this pseudocode lol):

```
%script{src: minifiable_asset_path('js/teacher_dashboard/index.js'), data: {selectedTab: @tab}}
```

Reply:
that way, our component doesn't need to do any URL-parsing logic